### PR TITLE
Migrate doku - Plugin quick guid

### PR DESCRIPTION
**Pull request by swDennis**

### CODE REVIEW

File: source/developers-guide/plugin-quick-start/index.md [hunk 1]
```diff
@@ -24,11 +24,13 @@ On the left you can see Shopware's default directory structure, as you will find
 
 **build**: (Not in release packages) Contains our ant-based build engine with various scripts to e.g. install dependencies, reset the database etc. When installing Shopware from GitHub you want to run `ant build-unit` first. If you use the install package, you can just use our web-based installer
 
+**custom/plugins** This directory contains all plugin for shopware 5.2 and above.
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: This directory contains all plugins based on the plugin system introduced with Shopware 5.2

File: source/developers-guide/plugin-quick-start/index.md [hunk 2]
```diff
@@ -117,195 +141,119 @@ In many cases you might want to modify the template. Shopware makes use of the S
...
  }
 ```
 
-3 things happen here:
- * we subscribe to an event
- * we call the createConfig method
- * we return a success flag
-
-In order to subscribe to an event, we use the `$this->subscribeEvent` helper method from the bootstrap base class. The first parameter is the event to subscribe to, in this case `Enlight_Controller_Action_PostDispatchSecure_Frontend`. As mentioned above, this event will be triggered every time **after** a controller in the frontend was executed. The second parameter is the callback function that should be called - `onFrontendPostDispatch`
+It's important to extend the `Shopware\Components\Plugin` class, as it holds a lot of helper functions you will need for plugin development.
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: actually the base class doesn't not contain "many" helpers :see_no_evil:

File: source/developers-guide/plugin-quick-start/index.md [hunk 3]
```diff
@@ -117,195 +141,119 @@ In many cases you might want to modify the template. Shopware makes use of the S
...
 
-    $view->addTemplateDir(
-        __DIR__ . '/Views'
-    );
+       $view->addTemplateDir($this->pluginDirectory . '/Resources/views');
 
-    $view->assign('slogan', $this->getSlogan());
-    $view->assign('sloganSize', $this->Config()->get('font-size'));
-    $view->assign('italic', $this->Config()->get('italic'));
+       $view->assign('fontSize', $this->config['fontSize']);
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: @philipgatzka changed theses variables in the example plugins
see here https://github.com/shopware/devdocs/pull/729

File: source/developers-guide/plugin-quick-start/index.md [hunk 4]
```diff
@@ -315,35 +263,74 @@ Finally we want to assign some configuration to the template - in this case a sl
...
+    <elements>
+        <element type="text" scope="locale">
+            <name>slogan</name>
+            <label lang="de">Dein Slogan</label>
+            <label lang="en">Your slogan</label>
+            <description lang="de">Dieser Slogan wird in in der Storefront angezeigt</description>
+            <description lang="en">This slogan will be shown on the storefront</description>
+        </element>
+        <element type="select" scope="locale">
+            <name>fontSize</name>
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: the names have also changed. see above

File: source/developers-guide/plugin-quick-start/index.md (hunk 1 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: d

File: source/developers-guide/plugin-quick-start/index.md (hunk 2 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: d

File: source/developers-guide/plugin-quick-start/index.md (hunk 3 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: d

File: source/developers-guide/plugin-quick-start/index.md [hunk 5]
```diff
@@ -315,35 +263,74 @@ Finally we want to assign some configuration to the template - in this case a sl
...
+            <label lang="de">Kursiv</label>
+            <label lang="en">Italic</label>
+            <value>false</value>
+        </element>
+    </elements>
+</config>
 ```
 
-In this case two config elements are added - a `select` element with the name "font-size" which will draw a combobox in the plugin configuration. The content of the combobox is defined in the "store" array element. Additionally we are able to have a label for the configuration using the "label" key and a default value using the "value" key.
+In this case two config elements are added - a `select` element with the name "font-size" which will draw a combobox in the plugin configuration. The content of the combobox is defined in the "store" element.
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: with the name "swagSloganFontSize"

File: source/developers-guide/plugin-quick-start/index.md [hunk 6]
```diff
@@ -315,35 +263,74 @@ Finally we want to assign some configuration to the template - in this case a sl
...
+            <label lang="en">Italic</label>
+            <value>false</value>
+        </element>
+    </elements>
+</config>
 ```
 
-In this case two config elements are added - a `select` element with the name "font-size" which will draw a combobox in the plugin configuration. The content of the combobox is defined in the "store" array element. Additionally we are able to have a label for the configuration using the "label" key and a default value using the "value" key.
+In this case two config elements are added - a `select` element with the name "font-size" which will draw a combobox in the plugin configuration. The content of the combobox is defined in the "store" element. 
+Additionally we are able to have a label for the configuration using the "label" property and a default value using the "value" property.
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: ... we ar able to define a label ...

File: source/developers-guide/plugin-quick-start/index.md [hunk 7]
```diff
@@ -431,3 +419,107 @@ After clearing the cache, your frontend might look like this:
 ![Slogan of the day in the frontend](img/result.png)
 
 You can find a installable ZIP package of this plugin <a href="{{ site.url }}/exampleplugins/SwagSloganOfTheDay.zip">here</a>.
+
+
+### The plugin base file methods
+In the next case we create a simple plugin that extend the s_articles_attributes table using the attribute crud service.
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: ... plugin that extends the ...

File: source/developers-guide/plugin-quick-start/index.md [hunk 8]
```diff
@@ -431,3 +419,107 @@ After clearing the cache, your frontend might look like this:
...
+ class SwagQuickStart extends Plugin
+ {
+     public function install(InstallContext $installContext)
+     {
+         // create a new attribute using the attribute crud service
+         $attributeCrudService = $this->container->get('shopware_attribute.crud_service');
+         $attributeCrudService->update(
+            's_articles_attributes',
+            'quick_start',
+            'string'
```
Comment (mitelg) [inline source/developers-guide/plugin-quick-start/index.md]: we should use the constants here

File: source/developers-guide/plugin-quick-start/index.md (hunk 5 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: D

File: source/developers-guide/plugin-quick-start/index.md (hunk 6 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: D

File: source/developers-guide/plugin-quick-start/index.md (hunk 7 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: d

Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: D

File: source/developers-guide/plugin-quick-start/index.md (hunk 8 above)
Comment (swDennis) [reply source/developers-guide/plugin-quick-start/index.md]: D

**Pull request merged**